### PR TITLE
test(swift): add unit test cases for extension functions

### PR DESCRIPTION
**Pull request by 5d**
*Issue #, if available:*

*Description of changes:*

- add unit test cases for extension functions

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_datastore/example/ios/unit_tests/GraphQLRequestExtensionTests.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
```
Comment (Jordan-Nelson) [inline packages/amplify_datastore/example/ios/unit_tests/GraphQLRequestExtensionTests.swift:6]: *nit:* for consistency with other headers in the repo

```suggestion
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
```

File: packages/amplify_datastore/example/ios/unit_tests/PublisherExtensionTests.swift [hunk 2]
```diff
@@ -0,0 +1,109 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
```
Comment (Jordan-Nelson) [inline packages/amplify_datastore/example/ios/unit_tests/PublisherExtensionTests.swift:6]:
```suggestion
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
```

File: packages/amplify_datastore/example/ios/unit_tests/PublisherExtensionTests.swift [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        Task {
+            subject.send(1)
+            subject.send(2)
+            subject.send(3)
+            subject.send(completion: .finished)
+        }
+
+        await fulfillment(of: [expectation1, expectation2, expectation3], timeout: 1)
```
Comment (Jordan-Nelson) [inline packages/amplify_datastore/example/ios/unit_tests/PublisherExtensionTests.swift:39]: The tests are failing in CI with the message 

```
❌  /Users/runner/work/amplify-flutter/amplify-flutter/packages/amplify_datastore/example/ios/unit_tests/PublisherExtensionTests.swift:39:15: cannot find 'fulfillment' in scope
```

**Review by Jordan-Nelson — COMMENTED** [review]
Changes look fine but the tests are failing in CI

File: packages/amplify_datastore/example/ios/unit_tests/PublisherExtensionTests.swift (hunk 3 above)
Comment (5d) [reply packages/amplify_datastore/example/ios/unit_tests/PublisherExtensionTests.swift:39]: Thanks for reviewing @Jordan-Nelson.

The API `fulfillment` mentioned in the error log was introduced in Xcode 14.3.

As discussed with @Equartey, to utilize the datastore v2, flutter will need to update the CI GitHub runner to `macOS-14` or `macOS-14-arm64`, as the minimum supported version for datastore v2 has been updated to Swift 5.9.

**Pull request merged**